.circleci/config.yml

```
@@ -0,0 +1,156 @@
+version: 2
+
+aliases:
+  - &main-filters
+    filters:
+      branches:
+        ignore: main
+
+jobs:
+  # Check that the git history is clean and complies with our expectations
+  lint-git:
+    docker:
+      - image: circleci/python:3.9-buster
+    working_directory: ~/learning-analytics-playground
+    steps:
+      - checkout
+      - run:
+          name: Check absence of fixup commits
+          command: |
+            ! git log | grep 'fixup!'
+      - run:
+          name: Install gitlint
+          command: |
+            pip install --user gitlint
+      - run:
+          name: Lint commit messages added to main
+          command: |
+            ~/.local/bin/gitlint --commits origin/main..HEAD
+
+  # Check that the CHANGELOG has been updated in the current branch
+  check-changelog:
+    docker:
+      - image: circleci/buildpack-deps:buster-scm
+    working_directory: ~/learning-analytics-playground
+    steps:
+      - checkout
+      - run:
+          name: Check that the CHANGELOG has been modified in the current branch
+          command: |
+            git whatchanged --name-only --pretty="" origin..HEAD | grep CHANGELOG
+
+  # Check that the CHANGELOG max line length does not exceed 80 characters
+  lint-changelog:
+    docker:
+      - image: debian:buster-slim
+    working_directory: ~/learning-analytics-playground
+    steps:
+      - checkout
+      - run:
+          name: Check CHANGELOG max line length
+          command: |
+            # Get the longuest line width (ignoring release links)
+            test $(cat CHANGELOG.md | grep -Ev "^\[.*\]: https://github.com/openfun" | wc -L) -le 80
+
+  e2e-test:
+    docker:
+      - image: cypress/included:4.0.2
+        entrypoint: cypress
+      - image: mysql:3.2
+        name: edx_mysql
+        environment:
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+          MYSQL_DATABASE: edxapp
+          MYSQL_USER: edxapp_user
+          MYSQL_PASSWORD: password
+      - image: mongo:3.2
+        name: edx_mongodb
+      - image: redis:4.0.10-alpine
+        name: edx_redis
+      - image: sj26/mailcatcher:latest
+        name: edx_mailcatcher
+      - image: ${EDXAPP_IMAGE:-fundocker/edxapp:dogwood.3-fun-2.0.0}
+        name: edx_lms
+        environment:
+          SERVICE_VARIANT: lms
+          DJANGO_SETTINGS_MODULE: lms.envs.fun.docker_run
+          EDX_API_KEY: FakeEdXAPIKey
+          CORS_ALLOW_CREDENTIALS: true
+          CORS_ALLOW_INSECURE: true
+          CROSS_DOMAIN_CSRF_COOKIE_DOMAIN: .local.dev
+          CROSS_DOMAIN_CSRF_COOKIE_NAME: edx_csrf_token
+          SESSION_COOKIE_NAME: edxcsrftoken
+          CORS_ORIGIN_ALLOW_ALL: true
+          FEATURES:
+            {
+              "AUTOMATIC_AUTH_FOR_TESTING": true,
+              "RESTRICT_AUTOMATIC_AUTH": false,
+              "ENABLE_OAUTH2_PROVIDER": true,
+              "ENABLE_CORS_HEADERS": true,
+              "ENABLE_CROSS_DOMAIN_CSRF_COOKIE": true,
+            }
```

Avoid mixing JSON and YAML syntax and stick to the latter:

```suggestion
          FEATURES:
              AUTOMATIC_AUTH_FOR_TESTING: true
              RESTRICT_AUTOMATIC_AUTH: false
              ENABLE_OAUTH2_PROVIDER: true
              ENABLE_CORS_HEADERS: true
              ENABLE_CROSS_DOMAIN_CSRF_COOKIE: true
```